### ISSUE TITLE
Clang incorrectly complains about default template argument not being reachable 

### ISSUE THREAD

**Issue by ilya-biryukov** [llvm/llvm-project#66255]
Run a compiler on this code:
```cpp
// --- foo.cppm
export module foo;

export struct partial {};
export template <class T, class Cat = partial> concept is_comparable = true;

// --- bar.cppm
export module bar;

import foo;
template <class T>
void func() requires is_comparable<T> {}
```

The exact commands are:
```sh
$ clang++ -std=c++20 --precompile foo.cppm
$ clang++ -std=c++20 --precompile bar.cppm -fmodule-file=foo.pcm
```

Expected results: code compiles with no errors.
Actual: Clang produces the following error when compiling the module `bar`:
```sh
bar.cppm:6:22: error: default argument of 'is_comparable' must be imported from module 'foo' before it is required
void func() requires is_comparable<T> {}
                     ^
foo.cppm:5:39: note: default argument declared here is not reachable
export template <class T, class Cat = partial> concept is_comparable = true;
                                      ^
1 error generated.
```

**Comment by ilya-biryukov** [comment]
I believe this particular bug is only applicable to older Clang versions (definitely present in clang 14).
I could not actually reproduce at head.

But we are seeing a similar issue with a generated module map we're using for libc++, I will keep this issue open for a few days while I'm trying to get a reduced reproducer.

**Comment by cor3ntin** [comment]
@ilya-biryukov can we close?
